packages/component-submission/server/use-cases/updateSubmissionUseCase.js

```diff
@@ -0,0 +1,60 @@
+const { pick } = require('lodash')
+
+function getManuscriptData(data) {
+  return pick(data, [
+    'meta',
+    'coverLetter',
+    'opposedSeniorEditorsReason',
+    'opposedReviewingEditorsReason',
+    'previouslyDiscussed',
+    'previouslySubmitted',
+    'cosubmission',
+    'opposedReviewersReason',
+    'submitterSignature',
+    'disclosureConsent',
+  ])
+}
+
+function getEditorTeams(data) {
+  return pick(data, [
+    'suggestedSeniorEditors',
+    'opposedSeniorEditors',
+    'suggestedReviewingEditors',
+    'opposedReviewingEditors',
+  ])
+}
+
+function getReviewerTeams(data) {
+  return pick(data, ['suggestedReviewers', 'opposedReviewers'])
+}
+
+const initialize = ({ submission, logger }) => ({
+  execute: async (manuscriptId, userId, data) => {
+    const manuscriptData = getManuscriptData(data)
+    const editorTeams = getEditorTeams(data)
+    const reviewerTeams = getReviewerTeams(data)
+
+    try {
+      await submission.initialize(manuscriptId, userId)
+
+      await submission.updateManuscript(manuscriptData)
+      await submission.updateAuthorTeam(data.author)
+      await submission.updateEditorTeams(editorTeams)
+      await submission.updateReviewerTeams(reviewerTeams)
+    } catch (error) {
+      const expected = 'Data Integrity Error'
+      if (error.message.startsWith(expected)) {
+        logger.error(`Expected a ${expected}, ${error.message}`)
+      } else {
+        // not an error we were expecting.
```

Comment:
Can probably remove this comment now https://github.com/elifesciences/elife-xpub/pull/2009/files#diff-5b518139d4766e3a75dc9c2cb1213c90R86 and https://github.com/elifesciences/elife-xpub/pull/2009/files#diff-5b518139d4766e3a75dc9c2cb1213c90R122 are expected error cases given 'bad' `data` objects.